### PR TITLE
feat(compiler): add nil

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Adds support for nil assignment to optional types

As described: https://github.com/winglang/wing/issues/436

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingc/src/jsify.rs [hunk 1]
```diff
@@ -327,6 +327,7 @@ impl<'a> JSifier<'a> {
 				}
 			}
 			ExprKind::Literal(lit) => match lit {
+        Literal::Nil => "null".to_string(),
```
Comment (eladb) [inline libs/wingc/src/jsify.rs:330]:
```suggestion
        Literal::Nil => "undefined".to_string(),
```

**Review by eladb — APPROVED** [review]
Nice!

**Review by Chriscbr — CHANGES_REQUESTED** [review]

File: examples/tests/invalid/nil.w [hunk 2]
```diff
@@ -0,0 +1,22 @@
+bring cloud;
+
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply examples/tests/invalid/nil.w:2]: Got it. I feel like nil types is something we should avoid in the language if there isn't a real world use case for it (minimize language surface area etc etc.), but seeing as `nil` is indeed a type in the spec and @eladb has already blessed this PR, I can only assume he's in favor of it, so let's try it out. 🙂

**Review by Chriscbr — APPROVED** [review]

File: examples/tests/invalid/nil.w (hunk 2 above)
[the remark this answers is not in the document]
Comment (eladb) [reply examples/tests/invalid/nil.w:2]: @hasanaburayyan I think that there was a mixup in the spec (we need to fix it). @Chriscbr got it right -- `nil` is not a type, it's a value that can be assigned to any optional to represent "lack of value". But it shouldn't be possible to use it as a type (e.g. (`let x: nil`)).

**Pull request merged**